functions/archiveEvents/source.js

```diff
@@ -247,117 +364,152 @@ getOneDay = function(urls,day){
+			    	let startPos = 0;
+			    	let endPos = cntrlFile.Objects.length;
+			    	let newCount = response.length;
+			    	if (endPos+newCount > maxControlObjects) 
+			    		startPos = endPos+newCount - maxControlObjects - 1;
+	
+			    	for (i=startPos;i< endPos;i++)
+			    		newobj.push(cntrlFile.Objects[i]);
+			    	for (i=0;i< response.length;i++) {
+			    		newobj.push(response[i]);
+			    	}
+			    	cntrlFile.Objects = newobj;
+			    	//console.log(`Now ${newobj.length} Objects`);
+			    	//console.log(EJSON.stringify(cntrlFile));
+			    	let controlData = [];
+			    	controlData.push(cntrlFile);
+			        return writeS3Object(`Control-${prefix}`,controlData)
+			        .then(response => {
+			        	let duration = new Date();
+			             duration = (duration - timer)/1000;
+			             console.log(`Processed ${itemCount} days for ${prefix} through ${cntrlFile.lastRun} in ${duration}s`);
+			             dataDogEvent("Atlas Event Archiver finished ",`Processed ${itemCount} days for ${prefix} through ${cntrlFile.lastRun} in ${duration}s`,prefix);
+			            return(response);	
+			        });
+			    })
+			})
+			.catch(err => {
+				    //console.log("doit");
+					if (err instanceof StitchError) {
+						let obj = EJSON.stringify(err.message);
+						let isMine = (err.message instanceof ArchiveException);
+						console.log(`${obj}, ${isMine}, ${err.name}, ${err.statck}`);
+					}
+		    		//dataDogEvent("Atlas Event Archiver failed ",`Processing ${prefix}. Status: ${err}`,prefix,"error");
+		    		return Promise.reject(`doit: Error is:\n-->${err}`);
+		    });
+		  })
+		  .catch(err => {
+				if (err instanceof StitchError) {
+					let obj = EJSON.stringify(err.message);
+					console.log(`${obj}, ${isMine}, ${err.name}, ${err.statck}`);
+				}
+	    		dataDogEvent("Atlas Event Archiver failed ",`Processing ${prefix}. Status: ${err}`,prefix,"error");
+	    		console.error(`archiveEvents: Error is:\n-->${err}`);
+	    		return false
+		
+		  });
+	  }
+	  catch(error) {
+			if (error instanceof StitchError) 
+				console.log(EJSON.stringify(error));
+			else
+				console.log(`Outer Catch: ${error}`);
+			return(false); 
+	  }
 }
 
 exports = function(prefix_in){
-	let mco = context.values.get('maxControlObjects');
-	if (typeof(mco) == 'number') maxControlObjects = mco;
-	console.log(typeof(mco));
-	let mdp = context.values.get('maxDaysToProcess');
-	if (typeof(mdp) == 'number') maxDaysToProcess = mdp;
-	let s3b = context.values.get('s3Bucket');
-	if (typeof(s3b) == 'string') s3Bucket = s3b;
-	console.log(typeof(s3b));
-	let s3r = context.values.get('s3Region');
-	if (typeof(s3r) == 'string') s3Region = s3r;
-	console.log(`Settings: MCO=${maxControlObjects}, MDP=${maxDaysToProcess}, S3B=${s3Bucket}, S3R=${s3Region}.`);
-	let result = doit(prefix_in);
-	return (result);
+	try {
+		let mco = context.values.get('maxControlObjects');
+		if (typeof(mco) == 'number') maxControlObjects = mco;
```

Comment:
[nit] I would suggest declaring the defaults here instead of above so someone reading the code doesn't have to scroll to the top to find out what it is.